### PR TITLE
test: program invoking all OpCodes

### LINKED ISSUE #1646: test: program invoking all OpCodes

### PR DESCRIPTION
**Pull request by lorenzogentile404**

### CODE REVIEW

File: arithmetization/src/test/java/net/consensys/linea/zktracer/OpCodesTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @Test
+  public void AllOpCodesTest() {
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(getAllOpCodesProgram());
+    bytecodeRunner.run();
+  }
+
+  private Bytes getAllOpCodesProgram() {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+    for (OpCodeData opCodeData : opCodeToOpCodeDataMap.values()) {
+      if (opCodeData.instructionFamily() != InstructionFamily.HALT) {
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/OpCodesTest.java]: You should also remove the `JUMP` family

Comment (lorenzogentile404) [reply arithmetization/src/test/java/net/consensys/linea/zktracer/OpCodesTest.java]: Ok.

File: arithmetization/src/test/java/net/consensys/linea/zktracer/OpCodesTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          && opCodeData.instructionFamily() != InstructionFamily.JUMP) {
+        OpCode opCode = opCodeData.mnemonic();
+        int nPushes = opCodeData.stackSettings().delta();
+        int nPops = opCodeData.stackSettings().alpha();
+        for (int i = 0; i < nPushes; i++) {
+          program.push(0);
+        }
+        program.op(opCode);
+        for (int i = 0; i < nPops; i++) {
+          program.op(OpCode.POP);
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/OpCodesTest.java:51]: `alpha` is either 0 or 1. So the natural thing isn't a for loop, it's
```java
if (opCodeData.stackSettings().alpha() != 0) program.op(OpCode.POP);
```

Comment (lorenzogentile404) [reply arithmetization/src/test/java/net/consensys/linea/zktracer/OpCodesTest.java:51]: Done.

**Review by OlivierBBB — APPROVED** [review]

**Pull request merged**